### PR TITLE
cookiecutter base branch

### PR DESCRIPTION
**Pull request by callmestu**
# Contributor Comments

Generated a new base project with cc-python and copied the files into this branch. 

Notes:
- no CI
- If CI is added it will require tests for current code which will be double work because I plan to write tests for python/docker image refactor
- prefer to merge this before adding CI, then write the tests

## Pull Request Checklist

Thank you for submitting a contribution to `release_monitor`!

In order to streamline the review of your contribution we ask that you review and comply with the below requirements:

- [x] Rebase your branch against the latest commit of the target branch
- [x] If you are adding a dependency, please explain how it was chosen
- [x] If manual testing is needed in order to validate the changes, provide a testing plan and the expected results
- [x] If there is an issue associated with your Pull Request, align your PR
  title with [this documentation](https://help.github.com/en/articles/closing-issues-using-keywords)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -167,3 +167,4 @@ tags
 [._]*.un~
 
 # End of https://www.toptal.com/developers/gitignore/api/vim,python
+.DS_Store
```
Comment (JonZeolla) [inline .gitignore]: nit: Please put these at the beginning of the `.gitignore`, before the "Start" of the toptotal.com comment.  Just easier to check into at a skim

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+ENV LANG=C.UTF-8
+ENV LC_ALL=C.UTF-8
+ENV PYTHONDONTWRITEBYTECODE=1
+ENV PIP_NO_CACHE_DIR=1
+
+ARG VERSION
+ARG COMMIT_HASH
+
+LABEL org.opencontainers.image.title="release_monitor"
+LABEL org.opencontainers.image.description="temp"
```
Comment (JonZeolla) [inline Dockerfile]: Please fill out

**Comment by JonZeolla** [comment]
The `LICENSE` file is empty.  It should either be an agreed-on license, or "Copyright (c) 2020-2022, Seiso, LLC - All rights reserved."

File: Pipfile [hunk 3]
```diff
@@ -0,0 +1,19 @@
+[[source]]
+url = "https://pypi.org/simple"
+verify_ssl = true
+name = "pypi"
+
+[packages]
```
Comment (JonZeolla) [inline Pipfile:6]: This should include the files from https://github.com/SeisoLLC/release_monitor/blob/5cdddcca2b141878c91f60abb41f9185731f9719/requirements-to-freeze.txt and then update the `Pipfile.lock`

File: setup.cfg [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+parse = (?P<year>2[0-1]\d{2})\.(?P<month>(0\d|1[0-2]))(.(?P<increment>\d{2}))?
+serialize = {year}.{month}.{increment}
+commit_message = "Automatically generated release {new_version}"
+commit = True
+tag = True
+push = True
+
+[metadata]
+author = "Seiso"
+description = "temp"
```
Comment (JonZeolla) [inline setup.cfg]: Please fix this; mirror what's in the `Dockerfile`

File: main.py [hunk 5]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env python3
+"""
+release_monitor script entrypoint
+"""
+
+from release_monitor import config
+
+if __name__ == "__main__":
+    log = config.setup_logging()
+    raise NotImplementedError()
```
Comment (JonZeolla) [inline main.py:10]: Just to make sure we're on the same page, the idea is to get this PR merged before incorporating the logic from [lambda_function.py](https://github.com/SeisoLLC/release_monitor/blob/5cdddcca2b141878c91f60abb41f9185731f9719/lambda_function.py), right?  I am good with that.

**Review by JonZeolla — CHANGES_REQUESTED** [review]
See feedback; pretty close!

File: .gitignore (hunk 1 above)
Comment (callmestu) [reply .gitignore]: Makes sense.

**Comment by JonZeolla** [comment]
@callmestu is this still on your radar

File: main.py (hunk 5 above)
Comment (callmestu) [reply main.py:10]: @JonZeolla You are correct.

File: Dockerfile (hunk 2 above)
Comment (callmestu) [reply Dockerfile]: One of the least polished iterations ended up here!

File: Pipfile (hunk 3 above)
Comment (callmestu) [reply Pipfile:6]: Thanks. Done.

**Comment by callmestu** [comment]
It's back on my radar and all requested changes were made @JonZeolla!

File: Pipfile (hunk 3 above)
Comment (JonZeolla) [reply Pipfile:6]: @callmestu this doesn’t appear to be done

File: tasks.py [hunk 6]
```diff
@@ -0,0 +1,289 @@
...
+
+    # Identify and set the increment
+    for tag in reversed(REPO.tags):
+        if pattern.fullmatch(tag.name):
+            latest_release = tag.name
+            break
+    else:
+        latest_release = None
+
+    if latest_release and date_info == latest_release[:7]:
```
Comment (JonZeolla) [inline tasks.py]: This is using an old version of this with a bug. 

I suggest regenerating

https://github.com/SeisoLLC/cookiecutter-python/blob/eec275779cd02c083c308644e58bd532e75c8fd5/%7B%7Bcookiecutter.project_slug%7D%7D/tasks.py#L263

**Review by JonZeolla — CHANGES_REQUESTED** [review]
Is there a backlog task for CI?  If not let’s make one. Needs the goat at least

File: Pipfile (hunk 3 above)
Comment (callmestu) [reply Pipfile:6]: @JonZeolla I see it in this file: https://github.com/SeisoLLC/release_monitor/blob/92fc6589f3067b2c1fd739cce4e191d55de443c8/Pipfile
What am I missing?

File: tasks.py (hunk 6 above)
Comment (callmestu) [reply tasks.py]: This took a while to regenerate on python 3.9. It is now the latest rendered cc-python slug.

**Comment by callmestu** [comment]
> Is there a backlog task for CI? If not let’s make one. Needs the goat at least

Backlog issue #13 created to add CI goat ASAP.

**Comment by JonZeolla** [comment]
@callmestu why not just pull in the github actions from cookiecutter python?  I thought that was the point of this PR

**Comment by JonZeolla** [comment]
We have two choices.  We can either:
- [ ] Add CI including goat and testing.  Ensure that the goat passes
- [ ] Add tests from cookiecutter-python and merge them with the `lambda_function.py` logic to reduce duplication of argparsing, etc.
- [ ] Make sure .github has other important files like `CODEOWNERS`
- [ ] Provide a line-by-line list of what isn't included from the generated cookiecutter for validation/discussion.
- [ ] Remove the legacy `Makefile` and `requirements*.txt` files.

Or we can create a new project from scratch, and then migrate the business logic into it.

I suggest the latter; dealer's choice.  If you agree, we will need to force push the `main` branch; coordinate with me to do so

**Pull request closed without merging**